org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/SolsticeScanMonitor.java

```diff
@@ -244,6 +261,34 @@ public void scanFinished() throws ScanningException {
 		} catch (Exception e) {
 			throw new ScanningException("Could not write scan duration to NeXus file", e);
 		}
+		
+		final long estimatedScanTimeMillis = model.getScanInformation().getEstimatedScanTime();
+		Duration scanDeadTime = scanDuration.minus(estimatedScanTimeMillis, ChronoUnit.MILLIS);
+		String scanDeadTimeStr = durationInMillisToString(scanDeadTime);
+		final Dataset scanDeadTimeDataset = DatasetFactory.createFromObject(scanDeadTimeStr);
+		try {
+			this.scanDeadTime.setSlice(null, scanDeadTimeDataset,
+					new int[] { 0 }, new int[] { 1 }, new int[] { 1 });
+		} catch (Exception e) {
+			throw new ScanningException("Could not write scan dead time to NeXus file", e);
+		}
+		
+		double deadTimePercent = ((double) scanDeadTime.toMillis() / scanDuration.toMillis()) * 100.0;
+		final String deadTimePercentStr = String.format("%.2f", deadTimePercent);
+		final Dataset deadTimePercentDataset = DatasetFactory.createFromObject(deadTimePercentStr);
+		try {
+			this.scanDeadTimePercent.setSlice(null, deadTimePercentDataset,
+					new int[] { 0 }, new int[] { 1 }, new int[] { 1 });
+		} catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00108)